src/charm.py

```diff
@@ -449,6 +453,20 @@ def _prometheus_layer(self):
 
         return layer
 
+    def _patch_k8s_service(self):
+        """Fix the Kubernetes service that was setup by Juju with correct port numbers"""
+        if self.unit.is_leader() and not self._stored.k8s_service_patched:
```

Comment:
So every time the leader changes, k8s port will be patched again. Is this the required behaviour ?

Reply:
In that case it would effectively be a no-op, cuz there is nothing to patch :-)

Reply:
It would be nice though to have the patch logic check if it is already patched instead of removing and creating a new service each time.